Fehler beim Kompilieren mit aktivem Webdebug.

collect2.exe: error: ld returned 1 exit status
exit status 1
Fehler beim Kompilieren für das Board Generic ESP8266 Module.

`c:/users/michael/documents/arduinodata/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.0.4-gcc10.3-1757bed/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: sketch\ShineWiFi-ModBus.ino.cpp.o:D:\Ablage\01_smart home\Growatt\GrowattShine_neues Projekt\Growatt_ShineWiFi-S-master\SRC\ShineWiFi-ModBus/WebDebug.h:5: multiple definition of `acWebDebug'; sketch\ShineMqtt.cpp.o:D:\Ablage\01_smart home\Growatt\GrowattShine_neues Projekt\Growatt_ShineWiFi-S-master\SRC\ShineWiFi-ModBus/WebDebug.h:5: first defined here
c:/users/michael/documents/arduinodata/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.0.4-gcc10.3-1757bed/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: sketch\ShineWiFi-ModBus.ino.cpp.o:D:\Ablage\01_smart home\Growatt\GrowattShine_neues Projekt\Growatt_ShineWiFi-S-master\SRC\ShineWiFi-ModBus/WebDebug.h:6: multiple definition of `u16WebMsgNo'; sketch\ShineMqtt.cpp.o:D:\Ablage\01_smart home\Growatt\GrowattShine_neues Projekt\Growatt_ShineWiFi-S-master\SRC\ShineWiFi-ModBus/WebDebug.h:6: first defined here`

--- a/SRC/ShineWiFi-ModBus/ShineMqtt.h
+++ b/SRC/ShineWiFi-ModBus/ShineMqtt.h
@@ -3,8 +3,8 @@
 #include "Config.h"
 
 #if MQTT_SUPPORTED == 1
-    #include "Arduino.h"
-    #include <ShineWifi.h>
+    #include <Arduino.h>
+    #include "ShineWifi.h"
     #include <PubSubClient.h>
     #include <stdbool.h>
     #include "WebDebug.h"

--- a/SRC/ShineWiFi-ModBus/WebDebug.c
+++ b/SRC/ShineWiFi-ModBus/WebDebug.c
@@ -1,0 +1,5 @@
+#include "WebDebug.h"
+#include <stdint.h>
+
+char acWebDebug[1024] = "";
+uint16_t u16WebMsgNo = 0;

--- a/SRC/ShineWiFi-ModBus/WebDebug.h
+++ b/SRC/ShineWiFi-ModBus/WebDebug.h
@@ -2,8 +2,8 @@
 #define GROWATT_SHINEWIFI_S_WEBDEBUG_H
 
 #if ENABLE_WEB_DEBUG == 1
-char acWebDebug[1024] = "";
-uint16_t u16WebMsgNo = 0;
+extern char acWebDebug[1024];
+extern uint16_t u16WebMsgNo;
 #define WEB_DEBUG_PRINT(s) {if( (strlen(acWebDebug)+strlen(s)+50) < sizeof(acWebDebug) ) sprintf(acWebDebug, "%s#%i: %s\n", acWebDebug, u16WebMsgNo++, s);}
 #else
 #undef WEB_DEBUG_PRINT

